redbot/cogs/mod/settings.py

```diff
@@ -87,35 +99,131 @@ async def hierarchy(self, ctx: commands.Context):
                 _("Role hierarchy will be ignored when moderation commands are issued.")
             )
 
-    @modset.command()
+    @modset.group()
+    @commands.guild_only()
+    async def mentionspam(self, ctx: commands.Context):
+        """
+        Manage the automoderation settings for mentionspam.
+        """
+
+    @mentionspam.command(name="warn")
+    @commands.guild_only()
+    async def mentionspam_warn(self, ctx: commands.Context, max_mentions: int = 0):
+        """
+        Sets the autowarn conditions for mention spam.
+
+        Users will be warned if they send any messages which contain more than
+        `<max_mentions>` mentions.
+
+        `<max_mentions>` Must be 0 or greater. Set to 0 to disable this feature.
+        """
+        mention_spam = await self.config.guild(ctx.guild).mention_spam.all()
+        if not max_mentions:
+            if not mention_spam["warn"]:
+                return await ctx.send(_("Autowarn for mention spam is already disabled."))
+            await self.config.guild(ctx.guild).mention_spam.warn.set(False)
+            return await ctx.send(_("Autowarn for mention spam disabled."))
+
+        if max_mentions < 1:
+            return await ctx.send(_("`<max_mentions>` must be 1 or higher to autowarn."))
+
+        mismatch_message = ""
+
+        if mention_spam["kick"]:
+            if max_mentions >= mention_spam["kick"]:
+                mismatch_message += _("\nAutowarn is equal to or higher than autokick.")
```

Comment:
The "mismatch" messages should explain better what they mean. On the one hand, a user should know that their `warn` will usually not trigger since `ban` is the same value, however there are cases where it is good to have them at the same value as backups (if a ban fails, it will still warn).

Reply:
Couldn't think of a good message to explain, so kept it simple in hopes someone could help write a better explanatory.